src/CredentialsLoader.php

```diff
@@ -169,4 +169,13 @@ public function updateMetadata(
     $metadata_copy[self::AUTH_METADATA_KEY] = array('Bearer ' . $result['access_token']);
     return $metadata_copy;
   }
+
+  /**
+   * Gets the underlying `OAuth2` instance used to conduct authorization.
+   *
+   * @return OAuth2
+   */
+  public function getAuth() {
```

Comment:
This is a tough one. Currently the way to do this (and I'm not saying it's good) would be to send in a `CacheInterface` object. Any access token fetched would then be stored in the cache object (although the methods to retrieve it require `getCacheKey`, which is private).

In the [README](https://github.com/google/google-auth-library-php#call-the-apis) workflow, we don't ever obtain the `CredentialsLoader` object. It might make more sense to have a method on the `Middleware` classes to get the current access token from the cache. The cache would then need to be updated to set not just the token, but the expiry, etc.

Or we can just expose `OAuth2`, but again this would require you to call `ApplicationDefaultCredentials::getCredentials` and then create the Middleware or Subscriber yourself. Also it exposes the library's class structure, which shouldn't really be necessary.